src/coreclr/jit/emitarm64.cpp

```diff
@@ -6906,7 +6924,58 @@ void emitter::emitIns_R_R_R_I(instruction ins,
     }
     assert(fmt != IF_NONE);
 
-    instrDesc* id = emitNewInstrCns(attr, imm);
+    // An "instrDesc" will *always* be required.
+    // Under normal circumstances the instruction
+    // will be added to the emitted group. However,
+    // this is not correct for instructions that
+    // are going to overwrite already-emitted
+    // instructions and we therefore need space to
+    // hold the new instruction descriptor.
+    instrDesc* id;
+
+    // One cannot simply instantiate an instruction
+    // descriptor, so this array will be used to
+    // hold the instruction being built.
+    unsigned char tempInstrDesc[sizeof(instrDesc)];
```

Comment:
Given my above comment, all this code (and more) could be removed.

Reply:
This is now removed, following implementation of your suggestion to remove the last emitted instruction.